doc/ERROR_CONCEPT.md

```diff
@@ -0,0 +1,250 @@
+
+### Line 4: Reason
+
+The most relevant line out of all, no question to keep it or not.
+Still I would remove the prefix `Reason: ` because of the reduced error message length in the new format. It makes it clear
+that this line is the reason. Probably we could substitute it with the plugin which called the error: 
+`<plugin>: Could not set ...`
+
+### Line 3: Description
+
+Reasons to keep it:
+* Gives users a quicker overview about the error (eg. permission error/ resource error/ etc. or another type of categorization such as Markus
+suggested with temporary/permanent/conflict/etc.)
+* Can indicate the type of solution a user has to approach
+
+Reasons against it:
+* Extra line
+* What to do with "uncategorizable" errors
+* Maintenance effort: introduce a new category once there are enough similar errors and change all old errors to this category?
+* Awareness of developers: All devs have to know all categories and choose the correct one. Devs should also be prohibited to throw their error
+into the "uncategorized" category due to lazyness.
+
+Suggestion:
+This line is by far the most discussable one. I thought about categorizations such as
+* Permission/ Resource/ Parse/ External(C++, Ruby, Java)/.../ Miscellaneous:
+Most errors are resource errors (could not stat, no connection, could not close/open/find etc.) This again would not give users
+a particularly useful error indication since he/she has to lookup in the reason anyway. Even subdividing the `Resource` category
+would lead to the same problem. No matter how I look at it, the user
+will have to read the concrete reason in most cases which makes the Description line not that valuable any longer.
+* temporary/permanent/conflict/validation:
+This seems to be a better approach because it includes a possible solution for the user. How to categorize certain errors though
+seems impossible because there can be multiple reasons where to categorize an error. Take `Could not open file` as an example:
+if it is on a NFS and it is offline, *temporary* would be correct. If the file has too restrictive permissions, *permanent* would be more
+appropriate since the admin has to fix it. If the file is currently in use and semlocked by the system, *conflict* would be appropriate.
+* core kdb error/plugin error:
+Not much value for the user
+* differentiation to core features of elektra: Specification/ Notification/ KDB Core/ Bindings/ ...
+An error is again difficult to put into one bucket since some errors might be affected by multiple features (could not stat file for example
+which is used by multiple features).
+This approach would lead to the same problem as we have now (many errors, difficult to maintain)
+
+Personally I do not see how the benefit of a categorization outweighs the drawbacks because in by far the most cases users will have to look
+into the reason in more detail anyway. It should be enough for users to deal with the error just by the reason.
```

Comment:
The categorization is for applications: if it is a conflict, they can merge and kdbSet again, if the problem is a validation error, they might drop the key and try again; if the error is temporary, they might retry (retry is quite problematic though as it is unclear how often to retry and so on...)

Reply:
I think such solution approaches should be indicated by the "Solution: " part in the error message and not via a generic description. Users will have to look into the reason in more detail, no matter the conflict/validation error/ etc.
Therefore I do not see a big advantage of the `Description` field anymore and a possible categorization.